### PR TITLE
Gui: Update splashscreen to use multiple images

### PR DESCRIPTION
**Pull request by qewer33**
Added new splashscreens that showcases addons and models from different creators. Made the splashcreen randomly select one at startup. Also disabled the splashscreen messages (it's unnecessary + disabling it GREATLY improves startup time).

### CODE REVIEW

File: src/Gui/Icons/freecadsplash2x.png
Comment (maxwxyz) [inline src/Gui/Icons/freecadsplash2x.png:1]: ths is the only one where the image does not overlap the rectangle, is this by design?

**Comment by maxwxyz** [comment]
Some images seem to be pixelated and have a white border
grafik

This one is also pixelated and has a white border. Also here, the image including shadow is going to the edge of the image which results in a sharp border at the bottom. The idea of the shadow should be, that it slowly fades out towards the edge of the image so there are at least full transparent pixels before there is the border of the image: 
grafik


This one seems blurred and very dark:
grafik


Also, still not sure about promoting some addons directly, @chennes there was a similar discussion for the start page, which addons should be featured?

**Comment by qewer33** [comment]
@maxwxyz You're looking at the 2x versions while zoomed in. Those problems on the normal versions are barely noticeable and don't pose any visual problems. I agree not all of them are perfect but there's sadly no way I can get a perfect borderless backgroundless png of all (or any really) of them. Also the reason some of them are pixelated is because they were screenshotted by their creators without AA on (which is the default for FreeCAD I believe, or used to be at least).

We can ofc remove the ones that are bad but I don't think any of them are really that problematic as I previously said.

**Comment by qewer33** [comment]
Also I personally don't think promoting "some" addons is a problem. Those addons are clearly well-made and have a lot of users, plus they fill gaps in the FreeCAD workflow which new users might find useful. And having them there indirectly promotes other addons as well since it incentivizes the user to go to the Addon Manager.

File: src/Gui/MainWindow.cpp [hunk 1]
```diff
@@ -2019,8 +2021,13 @@ QPixmap MainWindow::splashImage() const
     if (fi.isFile() && fi.exists())
         splash_image.load(fi.filePath(), "PNG");
 
-    // if no image was found try the config
+    // set a random splash path
+    // TODO: is there any way to not hardcode this?
+    int random = rand() % 14; // 14 is splash count
```
Comment (luzpaz) [inline src/Gui/MainWindow.cpp:2026]: How can you find out the current existing number of splash images exist and use that as a variable?

Comment (chennes) [reply src/Gui/MainWindow.cpp:2026]: Start a counter at zero, start trying to access those resources, the first one that fails is the end of the count.

**Comment by chennes** [comment]
@qewer33 can you please confirm that we have explicit permission each of these images and projects?

**Comment by qewer33** [comment]
> @qewer33 can you please confirm that we have explicit permission each of these images and projects?

I've asked all the model creators for explicit permission (and they have confirmed)  *except* the FOSS projects (Thor, SnakeOil, LumenPnP, M2) and addons (but we don't need permissions for addons right? the images are from the wiki). I figured we could use their images (I screenshotted some myself from their files too) since they're FOSS + we're crediting them anyway. But if it's a problem I can open issues on their repos and ask the maintainers.

**Comment by luzpaz** [comment]
> OSS projects (Thor, SnakeOil, LumenPnP, M2

@qewer33 i bumped 3 out of 4 of them on twitter. Don't know who M2 belongs to.

**Comment by qewer33** [comment]
> i bumped 3 out of 4 of them on twitter. Don't know who M2 belongs to.
Thanks! I actually pinged the M2 guy on Discord a couple days ago but he hasn't replied.

**Comment by chennes** [comment]
Thanks! I think as a matter of courtesy we should try to reach out to everyone involved so they know we are showcasing their work (and if they object of course we will remove it).

**Comment by ChipCE** [comment]
Hi, I'm Chip from SnakeOil 3d printer project. 
Please go ahead. If there is something I can help, fell free to ping me.

**Comment by 3x380V** [comment]
What is the reason to change image size?
It would be also nice to have clean commit history. You can pick it from [3x380V:splash](https://github.com/3x380V/FreeCAD/tree/splash) branch (a side product of testing plus some cleanup commits from my attempt #15435)

**Comment by qewer33** [comment]
> What is the reason to change image size?

It should be pretty close to the older image size anyway, if it poses a problem I can resize them but they're fine imo.

> It would be also nice to have clean commit history. You can pick it from [3x380V:splash](https://github.com/3x380V/FreeCAD/tree/splash) branch (a side product of testing plus some cleanup commits from my attempt #15435)

I'll take a look later, thanks.

**Comment by luzpaz** [comment]
:wave: @angellm would you give us consent to use Thor as one of the splashscreen images in the upcoming 1.0 release ?

:wave: @sphawes would you give us consent to use LumenPnP as one of the splashscreen images in the upcoming 1.0 release ?

**Comment by AngelLM** [comment]
> 👋 @AngelLM would you give us consent to use Thor as one of the splashscreen images in the upcoming 1.0 release ?
> 
> 👋 @sphawes would you give us consent to use LumenPnP as one of the splashscreen images in the upcoming 1.0 release ?

Absolutely! It would be a great honour! 😄

**Comment by sphawes** [comment]
> 👋 @sphawes would you give us consent to use LumenPnP as one of the splashscreen images in the upcoming 1.0 release ?

Absolutely! Thank you so much for considering us for the splashscreen! 

The best CAD image to use is of the newest version (which is being released very soon) that you can find here: https://github.com/opulo-inc/lumenpnp/blob/rev05-mobo/img/v4-cad-hero.png

Please let me know if you'd like a different resolution, file format, etc.

**Comment by Andor** [comment]
It would be rather cool if projects on the splash screen will be bundled as examples. What do you think?

**Comment by qewer33** [comment]
Alright I cleaned up the commit history (also @3x380V I cherry-picked your fix commits as well), we have permissions for every model in this PR (other than the Fasteners, SheetMetal and Gears addons but their images are taken from the wiki anyway) so it should be ready for merging now @chennes.

Also this is just the initial PR for the initial RC, I will open new PRs that add more splashes (and maybe a dark/light splash theme system) for later RCs and the actual release.

File: src/Gui/Icons/freecadsplash11.png
Comment (maxwxyz) [inline src/Gui/Icons/freecadsplash11.png:1]: should use the Gears WB icon: https://github.com/looooo/freecad.gears/blob/master/freecad/gears/icons/gearworkbench.svg

File: src/Gui/Icons/freecadsplash11_2x.png
Comment (maxwxyz) [inline src/Gui/Icons/freecadsplash11_2x.png:1]: same here https://github.com/looooo/freecad.gears/blob/master/freecad/gears/icons/gearworkbench.svg

**Comment by qewer33** [comment]
ok this is bad...

**Comment by qewer33** [comment]
Should be ready for merge now

**Comment by MisterMakerNL** [comment]
@qewer33 thx for picking this up they all look awesome!

**Comment by sphawes** [comment]
@qewer33 Could we use [this image](https://github.com/opulo-inc/lumenpnp/blob/rev05-mobo/img/v4-cad-hero.png) for the LumenPnP image instead? The one currently there is an old version of the machine. Happy to do any photoshop to have it fit the format if you'd like!

**Comment by qewer33** [comment]
> @qewer33 Could we use [this image](https://github.com/opulo-inc/lumenpnp/blob/rev05-mobo/img/v4-cad-hero.png) for the LumenPnP image instead? The one currently there is an old version of the machine. Happy to do any photoshop to have it fit the format if you'd like!

Oh I forgot about that, sorry. Can you take a screenshot from a more isometric angle (like the one on the current splash) perhaps? That would look nice, thanks!

**Comment by sphawes** [comment]
Thanks @qewer33! Here's a couple options, lmk if you want any tweaks!
lumen-more-iso
lumen-more-iso2

**Comment by qewer33** [comment]
Thank you @sphawes! Does this look good? I tweaked the colors slightly to make it pop a little better, I'll push it if you approve.

image

**Comment by sphawes** [comment]
Looks awesome @qewer33, thank you so much!!

**Comment by 3x380V** [comment]
> Should be ready for merge now

@qewer33, looks good, thank you. Could you squash and rebase? People does not seem to do it here on merge...

**Comment by qewer33** [comment]
> Could you squash and rebase? People does not seem to do it here on merge...

What's wrong with the current commit history? *Other than the fixup commits*.

**Comment by yorikvanhavre** [comment]
@qewer33 what's the status regarding author permissions? Do we have the permission of everybody? Is this ready for merge?

**Comment by qewer33** [comment]
> @qewer33 what's the status regarding author permissions? Do we have the permission of everybody? Is this ready for merge?

Yep! We have permissions for everything (other than the Sheet Metal, Gears and Fasteners addons as I mentioned but their images are taken from the wiki and they're inside the FreeCAD ecosystem so) and the PR is ready for merge.

**Comment by luzpaz** [comment]
> > i bumped 3 out of 4 of them on twitter. Don't know who M2 belongs to.
> 
> Thanks! I actually pinged the M2 guy on Discord a couple days ago but he hasn't replied.

@qewer33 can you get M2 project to consent on this ticket ?

**Comment by qewer33** [comment]
@luzpaz I removed it for the time being.

**Comment by 3x380V** [comment]
> What's wrong with the current commit history? _Other than the fixup commits_.

Other than that it is perfectly fine!

**Comment by qewer33** [comment]
@GeneGH only the messages are removed, the version text is still there.

**Comment by qewer33** [comment]
No... The images don't have the version number.

**Comment by maxwxyz** [comment]
perfect

**Comment by 3x380V** [comment]
@WandererFan, any reason for not resolving fixups on the merge?

**Comment by tibfel** [comment]
what about including progress bar / gauge (semi transparent to look better)? It would help if the loading gets stuck for some reason.

**Comment by 3x380V** [comment]
That would need to change underlying logic completely as it is not known in advance how many items need to be initialized.

**Comment by tibfel** [comment]
After testing the newest build, I realized that now it is so fast (when removing messages), that there is no need to include it.

**Comment by PaddleStroke** [comment]
@qewer33 do you have the original .svg for the background ? I wanted to add a splashscreen for the text tool (as offered there https://github.com/FreeCAD/FreeCAD/pull/22217#issuecomment-3975217030).

**Comment by maxwxyz** [comment]
I have a version but it was not the latest. I think we should do a new design for the next release which is easier to change considering we will have 3 releases a year now @FreeCAD/design-working-group

**Pull request merged**